### PR TITLE
Add metadata key for holding an arbitrary identification token

### PR DESCRIPTION
**Pull request by matheuscscp**
Part of the PRs for https://github.com/fluxcd/helm-controller/issues/678 alongside https://github.com/fluxcd/helm-controller/pull/681 and https://github.com/fluxcd/notification-controller/pull/518

### CODE REVIEW

**Comment by matheuscscp** [comment]
@hiddeco nice! That's indeed much better. Would the revision be included in this token, or kept as a separate thing as it is today?

**Comment by hiddeco** [comment]
That's up to the emitting controller to decide, the notification-controller should just consume it as part of the key that handles rate limiting.

**Comment by matheuscscp** [comment]
> That's up to the emitting controller to decide, the notification-controller should just consume it as part of the key that handles rate limiting.

Yep my question is specifically about what to do on the notification-controller side. Right now it includes the `revision` key from the event metadata in the rate limiting key. My question is: are we gonna replace the use of the `revision` key there, or just add the new one?

**Comment by hiddeco** [comment]
We would use both, as they represent different things (source being used versus declarative configuration).

**Comment by matheuscscp** [comment]
@hiddeco I updated the PR :ok_hand:

File: apis/event/v1beta1/metadata.go [hunk 1]
```diff
@@ -30,4 +30,9 @@ const (
 	// MetaCommitStatusUpdateValue is the value of MetaCommitStatusKey
 	// used to signal a Git commit status update.
 	MetaCommitStatusUpdateValue string = "update"
+	// MetaTokenKey is the key used to hold an arbitrary token whose contents
```
[the remark this answers is not in the document]
Comment (matheuscscp) [reply apis/event/v1beta1/metadata.go:33]: done!

**Review by hiddeco — APPROVED** [review]

**Comment by hiddeco** [comment]
@stefanprodan @darkowlzz please take a look.

File: apis/event/v1beta1/metadata.go [hunk 2]
```diff
@@ -30,4 +30,11 @@ const (
...
```
[the remark this answers is not in the document]
Comment (hiddeco) [reply apis/event/v1beta1/metadata.go:33]: I would maybe just put this one above `MetaCommitStatusKey`.

**Comment by matheuscscp** [comment]
> LGTM!
> 
> Please rebase and also maybe update the PR description to not accidentally close [fluxcd/helm-controller#678](https://github.com/fluxcd/helm-controller/issues/678) as I think that requires changes in helm-controller like in [fluxcd/helm-controller#681](https://github.com/fluxcd/helm-controller/pull/681) .

@darkowlzz I updated the PR description, please let me know if it works

**Review by stefanprodan — APPROVED** [review]
LGTM

**Pull request merged**